### PR TITLE
Move route and fix sign out bug on game page

### PR DESCRIPTION
**Pull request by shandyk18**

### CODE REVIEW

File: ubg/src/pages/Game.js [hunk 1]
```diff
@@ -233,13 +233,16 @@ function FavoriteButton({usersCollection, game}) {
...
+    {user} ?
+    (
+      <Button
+        variant='contained'
+        color='primary'
+        onClick={() => addFavorite(
+            userGames, usersCollection, game, favorite, setFavorite, user.uid)}>
+        {favorite ? 'Delete from favorites' : 'Add to favorites'}
+      </Button>
+    ) : ' '
```
Comment (danieltre23) [inline ubg/src/pages/Game.js:245]: I think instead of '  ' you can use null, because I think that way it actually renders '  '

Comment (shandyk18) [reply ubg/src/pages/Game.js:245]: alright I'll try that out

File: ubg/src/pages/Game.js [hunk 2]
```diff
@@ -250,17 +250,21 @@ function Description({usersCollection, game, createRoom}) {
  */
 function FavoriteButton({usersCollection, game}) {
   const user = useUser();
-  const userGames = useFirestoreDocData(usersCollection.doc(user.uid)).games;
+  const userGames = useFirestoreDocData(
+      usersCollection.doc((user && user.uid) || ' ')).games;
```
Comment (danieltre23) [inline ubg/src/pages/Game.js:254]: Do you need user && user.uid? Or can you just use user.uid?

Comment (shandyk18) [reply ubg/src/pages/Game.js:254]: I need that there because it'll fail trying to read user.uid

Comment (danieltre23) [reply ubg/src/pages/Game.js:254]: Yes, but if just have user.uid there and its null then its going to go with '  '

Comment (danieltre23) [reply ubg/src/pages/Game.js:254]: Screenshot 2020-08-05 at 11 54 42 AM

Comment (danieltre23) [reply ubg/src/pages/Game.js:254]: Something like that, I mean its just so you don't have to do the && it's just to get rid of one extra condition.

Comment (shandyk18) [reply ubg/src/pages/Game.js:254]: the issue is user is null already, so trying to access the uid of a null object causes problems.

**Pull request merged**